src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -89,7 +89,7 @@ private void SetGlobalErrorIfTrue(bool arg)
             _hasDeclarationErrors = hasDeclarationErrors;
             SetGlobalErrorIfTrue(hasDeclarationErrors);
 
-            if (emittingPdb)
+            if (emittingPdb || moduleBeingBuiltOpt?.EmitOptions.EmitDynamicAnalysisData == true)
```

Comment:
is "== true" needed here? Can this be just "if (emittingPdb || moduleBeingBuiltOpt?.EmitOptions.EmitDynamicAnalysisData)"